src/models/arellano_vfi.jl

```diff
@@ -0,0 +1,333 @@
+using QuantEcon: tauchen, MarkovChain, simulate
+
+
+# ------------------------------------------------------------------- #
+# Define the main Arellano Economy type
+# ------------------------------------------------------------------- #
+
+"""
+Arellano 2008 deals with a small open economy whose government
+invests in foreign assets in order to smooth the consumption of
+domestic households. Domestic households receive a stochastic
+path of income.
+
+##### Fields
+* `β::Real` : Time discounting parameter
+* `γ::Real` : Risk aversion parameter
+* `r::Real` : World interest rate
+* `ρ::Real` : Autoregressive coefficient on income process
+* `η::Real` : Standard deviation of noise in income process
+* `θ::Real` : Probability of re-entering the world financial sector after default
+* `ny::Int` : Number of points to use in approximation of income process
+* `nB::Int` : Number of points to use in approximation of asset holdings
+* `ygrid::Vector{Float64}` : This is the grid used to approximate income process
+* `ydefgrid::Vector{Float64}` : When in default get less income than process would otherwise dictate
+* `Bgrid::Vector{Float64}` : This is grid used to approximate choices of asset holdings
+* `Π::Array{Float64, 2}` : Transition probabilities between income levels
+* `vf::Array{Float64, 2}` : Place to hold value function
+* `vd::Array{Float64, 2}` : Place to hold value function when in default
+* `vc::Array{Float64, 2}` : Place to hold value function when choosing to continue
+* `policy::Array{Float64, 2}` : Place to hold asset policy function
+* `q::Array{Float64, 2}` : Place to hold prices at different pairs of (y, B')
+* `defprob::Array{Float64, 2}` : Place to hold the default probabilities for pairs of (y, B')
+"""
+immutable ArellanoEconomy
+    # Model Parameters
+    β::Float64
+    γ::Float64
+    r::Float64
+    ρ::Float64
+    η::Float64
+    θ::Float64
+
+    # Grid Parameters
+    ny::Int
+    nB::Int
+    ygrid::Array{Float64, 1}
+    ydefgrid::Array{Float64, 1}
+    Bgrid::Array{Float64, 1}
+    Π::Array{Float64, 2}
+
+    # Value function
+    vf::Array{Float64, 2}
+    vd::Array{Float64, 2}
+    vc::Array{Float64, 2}
+    policy::Array{Float64, 2}
+    q::Array{Float64, 2}
+    defprob::Array{Float64, 2}
+end
+
+"""
+This is the default constructor for building an economy as presented
+in Arellano 2008.
+
+##### Arguments
+* `β::Real` : Time discounting parameter
```

Comment:
Please revise to include the `;` denoting keyword argument and put the default in parens. So this line should read:

```
`;β::Real(0.953)` : Time discounting parameter
```